apps/web/src/app/(networks)/(non-evm)/stellar/_common/lib/services/sushi-stellar-service.ts

```diff
@@ -0,0 +1,392 @@
+import { QuoteService, type SwapQuote } from './quote-service'
+import { RouterService, type SwapRoute } from './router-service'
+import {
+  type AddLiquidityParams,
+  type SwapExactInputParams,
+  type SwapExactInputSingleParams,
+  SwapService,
+} from './swap-service'
+
+/**
+ * Main service for SushiSwap operations on Stellar
+ * This is the primary interface for interacting with the DEX
+ */
+export class SushiStellarService {
+  private swapService: SwapService
+  private quoteService: QuoteService
+  private routerService: RouterService
+
+  constructor() {
+    this.swapService = new SwapService()
+    this.quoteService = new QuoteService()
+    this.routerService = new RouterService(this.quoteService)
+  }
+
+  /**
+   * Add liquidity using Position Manager
+   * Automatically increases liquidity on existing position or creates new one
+   */
+  async addLiquidity(
+    userAddress: string,
+    params: AddLiquidityParams,
+    signTransaction: (xdr: string) => Promise<string>,
+    signAuthEntry: (entryPreimageXdr: string) => Promise<string>,
+  ): Promise<{ txHash: string; tokenId: number; liquidity: bigint }> {
+    // Convert string amounts to bigint
+    const amount0 = BigInt(
+      Math.floor(
+        Number.parseFloat(params.token0Amount) * 10 ** params.token0Decimals,
+      ),
+    )
+    const amount1 = BigInt(
+      Math.floor(
+        Number.parseFloat(params.token1Amount) * 10 ** params.token1Decimals,
+      ),
+    )
+
+    const deadline = BigInt(
+      params.deadline || Math.floor(Date.now() / 1000) + 300,
+    )
+
+    // Always fetch pool info from contract (no more dynamic import needed)
+    const poolConfig = await getPoolInfoFromContract(params.poolAddress)
+    if (!poolConfig) {
+      throw new Error('Pool config not found')
+    }
+
+    // Check if user has existing position for this pool with same tick range
+
+    // Fetch user positions
+    const positions = await positionService.getUserPositionsWithFees({
+      userAddress,
+    })
+
+    // Find position with matching pool tokens, tick range, AND fee tier
+    const existingPosition = positions.find((pos) => {
+      const tokensMatch =
+        (pos.token0 === poolConfig.token0.contract &&
+          pos.token1 === poolConfig.token1.contract) ||
+        (pos.token0 === poolConfig.token1.contract &&
+          pos.token1 === poolConfig.token0.contract)
+
+      const ticksMatch =
+        pos.tickLower === params.tickLower && pos.tickUpper === params.tickUpper
+
+      const feeMatches = pos.fee === poolConfig.fee
+
+      return tokensMatch && ticksMatch && feeMatches
+    })
+
+    if (existingPosition) {
+      // Increase liquidity on existing position - NO try/catch here!
+      const result = await increaseLiquidity({
+        tokenId: existingPosition.tokenId,
+        amount0Desired: amount0,
+        amount1Desired: amount1,
+        amount0Min: BigInt(0), // TODO: Add slippage protection
```

Comment:
todo?

Reply:
This is not needed and has been removed.